src/main.rs

```diff
@@ -45,28 +45,47 @@ async fn main() {
     }
     println!("Consul lock acquired successfully");
 
-    let nomad_tag = match nomad::fetch_and_parse_service_tags(&config).await {
-        Ok(tag) => tag,
+    println!("Fetching DNS tags from the services in Consul");
+    let dns_tags = match consul_client.fetch_service_tags().await {
+        Ok(tags) => tags,
         Err(e) => {
-            eprintln!("Failed to fetch Nomad DNS tags: {}", e);
+            eprintln!("Failed to fetch Consul DNS tags: {}", e);
             return;
         }
     };
-    println!("Nomad DNS tags fetched successfully");
+
+    println!(
+        "Consul DNS tags fetched successfully. Length: {}",
+        dns_tags.len()
+    );
 
     let dns_provider: Box<dyn DnsProviderTrait> = match config.dns_provider {
         DnsProvider::Hetzner(config) => Box::new(hetzner_dns::HetznerDns { config }),
     };
 
-    let result = dns_provider.update_or_create_dns_record(&nomad_tag).await;
+    // Update or create DNS record for every dns tag
+    let mut all_success = true;
+    for dns_tag in dns_tags {
+        match dns_provider.update_or_create_dns_record(&dns_tag).await {
+            Ok(_) => println!("DNS record updated or created successfully"),
+            Err(e) => {
+                eprintln!(
+                    "Failed to update or create DNS record for {}: {}",
+                    dns_tag.hostname, e
+                );
+                all_success = false;
+            }
+        }
+    }
 
     // Release Lock
```

Comment:
If a record we've created ends up disappearing from consul for whatever reason we want to delete it. To do this, what I think we should do is that we need to keep track of a hashmap `ProviderRecordId => DnsRecord` which we store in consul.

At the start of the loop we first fetch the current value from consul, then we loop over the records we parsed from service tags.  If the record exists in the state already we skip it.  If it doesn't exist we create the record and insert it into the state.

After we've created all new records we loop over all items in the state.  If we find any that exist in the state but does not exist in the collection of parsed records then we delete the record and delete it from the state.

And finally we serialize the state and store it in consul again.  This makes the service stateless, and allows any other instance to take over processing when it acquires the lock.